tracing-attributes/tests/instrument.rs

```diff
@@ -3,6 +3,12 @@ use tracing::Level;
 use tracing_attributes::instrument;
 use tracing_mock::*;
 
+#[instrument]
+fn repro_2294() {
+    #![allow(unused_variables)]
+    let i = 42;
+}
```

Comment:
same comment as above, it might be worth changing this test so that it'll also fail to compile if we mangle or ignore the attribute? not a blocker.